src/utils/Helpers.js

```diff
@@ -0,0 +1,28 @@
+function setButtonText(
+  button,
+  isLoading,
+  loadingText = "Saving...",
+  defaultText = "Save"
+) {
+  if (isLoading) {
+    button.textContent = loadingText;
+  } else {
+    button.textContent = defaultText;
+  }
+}
+
+export function handleSubmit(request, evt, loadingText = "Saving...") {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  const defaultText = submitButton.textContent;
+  setButtonText(submitButton, true, loadingText, defaultText);
+  request()
+    .then(() => {
+      evt.target.reset();
+    })
+    .catch(console.error)
+
+    .finally(() => {
+      setButtonText(submitButton, false, defaultText);
```

Comment:
`setButtonText` has 4 arguments, and the 3rd is `loadingText`, and the 4th is `defaultText`

```js
      setButtonText(submitButton, false, loadingText, defaultText);

```